### PR TITLE
Fix #2428: Add translations for 1.15.1

### LINKED ISSUE #2428: Add translations for 1.15.1
- [x] Push new strings
- [x] Pull new translations

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #2428 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveRewardsUI/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -622,6 +634,51 @@
...
+"SKUPurchaseBuyWithBAT" = "Kaufen mit %@";
+
+/* No comment provided by engineer. */
+"SKUPurchaseCompletedBody" = "Viel Spaß beim Einkauf.";
+
+/* No comment provided by engineer. */
+"SKUPurchaseCompletedTitle" = "Sie sind fertig!";
+
+/* First '%@' becomes 'Buy with BAT' or 'Buy with BAP' based on region. Second '%@' becomes 'Brave’s Terms of Sale' */
+"SKUPurchaseDisclaimerInformation" = "Indem Sie auf '%1$@' klicken, erklären Sie sich einverstanden mit den %2$@.";
```
Comment (kylehickinson) [inline BraveRewardsUI/de.lproj/Localizable.strings:662]: This is the correct syntax for this, right?

Comment (kylehickinson) [reply BraveRewardsUI/de.lproj/Localizable.strings:662]: Probably should double check that this works (not sure if regular `String(format:)` API takes these locations into account. May have to switch over to the Bundle.localizedFormat or w/e its called

Comment (kylehickinson) [reply BraveRewardsUI/de.lproj/Localizable.strings:662]: @iccub Tested and it works correctly

**Review by kylehickinson — APPROVED** [review]

File: BraveRewardsUI/de.lproj/Localizable.strings (hunk 1 above)
Comment (iccub) [reply BraveRewardsUI/de.lproj/Localizable.strings:662]: I tested and it works fine

**Pull request merged**